tensorboard/plugins/graph/graph_util.py

```diff
@@ -0,0 +1,214 @@
+      input_layer_name = _scoped_name(node_name, input_layer)
+      inbound_node_name = _scoped_name(name_scope, inbound_node[0])
+      input_to_in_layer[input_layer_name] = inbound_node_name
+  elif is_parent_functional_model and not is_functional_model:
+    # Sequential model can take only one input. Make sure inbound to the
+    # model is linked to the first layer in the Sequential model.
+    prev_node_name = _scoped_name(name_scope, inbound_nodes[0][0])
+  elif not is_parent_functional_model and prev_node_name and is_functional_model:
+    assert len(input_layers) == 1, (
+        'Cannot have multi-input Functional model when parent model '
+        'is not Functional. Number of input layers: %d' % len(input_layer))
+    input_layer = input_layers[0]
+    input_layer_name = _scoped_name(node_name, input_layer)
+    input_to_in_layer[input_layer_name] = prev_node_name
+
+  if is_functional_model and output_layers:
+    layers = _norm_to_list_of_layers(output_layers)
+    layer_names = [_scoped_name(node_name, layer[0]) for layer in layers]
+    model_name_to_output[node_name] = layer_names
+  else:
+    last_layer = layer_config.get('layers')[-1]
+    last_layer_name = last_layer.get('config').get('name')
+    output_node = _scoped_name(node_name, last_layer_name)
+    model_name_to_output[node_name] = [output_node]
+  return (input_to_in_layer, model_name_to_output, prev_node_name)
+
+
+def keras_model_to_graph_def(keras_layer):
+  """Returns GraphDef representation of the model.
+
+  Note that it only supports models that implemented get_config().
+
+  Args:
+    keras_layer: A dict from Keras model.to_json()
+
+  Returns:
+    A GraphDef representation of layers.
+  """
+  input_to_layer = {}
+  model_name_to_output = {}
+  g = GraphDef()
+
+  # Sequential model layers do not have a field "inbound_nodes" but
+  # instead of defined implicitly via order of layers.
+  prev_node_name = None
+
+  for (name_scope, layer) in _walk_layers(keras_layer):
+    if _is_model(layer):
+      (input_to_layer, model_name_to_output, prev_node_name) = _update_dicts(
+          name_scope, layer, input_to_layer, model_name_to_output, prev_node_name)
+      continue
+
+    layer_config = layer.get('config')
+    node_name = _scoped_name(name_scope, layer_config.get('name'))
+
+    node_def = g.node.add()
+    node_def.name = node_name
+
+    if layer.get('class_name') is not None:
+      keras_cls_name = layer.get('class_name').encode('ascii')
+      node_def.attr['keras_class'].s = keras_cls_name
+
+    if layer_config.get('dtype') is not None:
+      tf_dtype = dtypes.as_dtype(layer_config.get('dtype'))
+      node_def.attr['dtype'].type = tf_dtype.as_datatype_enum
+
+    if layer.get('inbound_nodes') is not None:
+      for maybe_inbound_node in layer.get('inbound_nodes'):
+        inbound_nodes = _norm_to_list_of_layers(maybe_inbound_node)
+        for [name, size, ind, _] in inbound_nodes:
```

Comment:
What is "ind"? I guess from reading the code it is index? Can you rename to that or something similarly descriptive?